Revisión de todos los archivos del repositorio

nnzgab opened the pull request:
Archivos completos para revisión

nnzgab committed:
Add files via upload

scaprile commented on README.md:
Bastante pobre y plagada de errores la redacción

nnzgab replied:
Muchas gracias por revisar mi TP, aprecio que se haya tomado el tiempo para proporcionarme retroalimentación.
Reconozco que la redacción no era la mejor y he trabajado para mejorarla.

scaprile commented on conexion.PNG:
Yo sólo sé leer diagramas esquemáticos, así que te perdiste que te hiciera comentarios sobre el circuito electrónico.

nnzgab replied:
Estoy trabajado en realizar un esquemático, y agregare lo antes posible al readme, seria muy valioso para mi sus comentarios.

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,331 @@
+        ledArm = ON;
+        ledReady = OFF;
+        ledSystem = OFF;
+    }
+    if(accumulatedTime >= 250 && accumulatedTime < 500){
```
`else if(accumulatedTime < 500)`

nnzgab replied:
gracias, eh aplicado la corrección

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,331 @@
+        ledArm = OFF;
+        ledReady = ON;
+        ledSystem = OFF;
+    }
+    if(accumulatedTime >= 500 && accumulatedTime < 750){
```
`else if(accumulatedTime < 750)`

nnzgab replied:
gracias, eh aplicado la corrección

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,331 @@
+        ledArm = OFF;
+        ledReady = OFF;
+        ledSystem = ON;
+    }
+    if(accumulatedTime == 750){
```
`else`

nnzgab replied:
gracias, eh aplicado la corrección

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,331 @@
+            ledSystem = 0;
+            accumulatedTime = 0;
+            timeout = 0;
+			delay(250);
```
NO, no, no

nnzgab replied:
se quitaron bloqueos de 250ms

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,331 @@
+    ledSystem = OFF;
+}
+
+void prenderLedNumero(int numeroLed){
+    switch(numeroLed){
+        case 1: 
+        ledArm = 1;
+        serial_port.write( "LedArm encendido\r\n", 18);
+        serial_port.write( "Presione Boton 1 para continuar\r\n", 33);
+        break;
+        case 2: 
+        ledReady = 1; 
+        serial_port.write( "LedReady encendido\r\n", 20);
+        serial_port.write( "Presione Boton 1 para continuar\r\n", 33);
+        break;
+        case 3: 
+        ledSystem = 1; 
+        serial_port.write( "LedSystem encendido\r\n", 21);
+        serial_port.write( "Presione Boton 1 para continuar\r\n", 33);
+        break;
+        default: 
+        break;
+    }
+}
```
Dependencia innecesaria, quién podría imaginarse que encender un LED tiene el efecto secundario de escribir la UART ?

nnzgab replied:
gracias, he quitado dependencia innecesaria

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,331 @@
+            currentLED++;
+        }
+    if (tecla_1 == 1) {
+        delay(250);
```
NO

nnzgab replied:
se quitaron bloqueos de 250ms

scaprile reviewed (changes requested):
Demasiado convoluted para algo tan sencillo.
Todo ese enjambre se resuelve con una FSM (máquina de estados finitos)
Por favor sacá esos bloqueos de 250ms

nnzgab committed:
se corrige secuenciaLed()

se trabaja sobre las indicaciones quitando partes redundantes.

nnzgab committed:
se agrega diagrama en bloque

se agrega un diagrama en bloque general

nnzgab committed:
mejora de contenido y redacción del README

se trabajo en la mejora del contenido y redacción del README

nnzgab committed:
se quita dependencia innecesaria

se quita dependencia innecesaria de escribir en la uart al encender un led

nnzgab committed:
se quitan los bloqueos de 250ms

se modifica startTest() recorrer_teclas()
se agregan funciones debounceTeclaInit() y debouceTecUpdate()
para quitar delay en controlarLed()

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,395 @@
+    if(runTest == false && endTest == false){
+	static bool button1Pressed = false;
+        static bool button2Pressed = false;
+        if(tecla_1 == 1){
```
Si leíste el comentario anterior, en este punto, ignoralo. Es una cosa particular de Mbed que sobrecarga un operador sólo para molestar a los que conocemos el resto del mundo.

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,395 @@
+            serial_port.write( "Presione Tecla 1 para continuar\r\n", 33);
+            currentLED++;
+    }else{
+	    bool pulsacionRealTec1 = debouceTecUpdate();
```
y a quién se le podría ocurrir que acá se iba a llamar al anti-rebote ?

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,395 @@
+float analogReadingVoltage(float voltajeVbat){
+    voltajeVbat = 3.3 * vbatReading;
+    return voltajeVbat;
+}
```
Qué sentido tiene pasar un parámetro que no usás ?

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,395 @@
+
+void checkTimeout(){
+    if(timerOutActivated == true){
+        timeout = timeout+1;
```
`++timeout`

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,395 @@
+int accumulatedTime = 0;
+int testNum = 0;
+
+float vbatReading = 0.0;
```
Global innecesaria que te tapa el moco que tenés en la función que lee el ADC

scaprile reviewed (changes requested):
Todavía no puedo seguir esto.
Es ejemplo paradigmático de `spaghetti coding`
Intentá darle forma sin tener que andar saltando constantemente de arriba a abajo para adivinar el funcionamiento.

nnzgab committed:
replanteo de gran parte del main.cpp

Se replantearon varias partes del codigo.
Se busco responder a las indicaciones.
	dar mas forma
	evitar saltos arriba y abajo
se quitaron funciones innecesarias.

scaprile commented on main.cpp:
```diff
@@ -37,359 +51,263 @@ DigitalOut outBuzzer(BUZZER);
 
-void inputsInit()
-{
+void inputsInit(){
```
Antes: estilo K&R, preferido por los viejitos y el linux kernel
Ahora: estilo preferido por los que no vienen de C.

Elegí uno y mantenelo, en main() usaste otro estilo.

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,313 @@
+
+//=====[Declaration and initialization of public global variables]=============
+
+testState_t current_State = PREVIOUS_TEST;
+testState_t nextState = PREVIOUS_TEST;
```
Qué sentido tiene inicializar estas variables globales que después volvés a inicializar en una función de inicialización llamada al inicializar ?

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,313 @@
+    outBuzzer = OFF;
+}
+
+void estadoInicial(){
+    current_State = PREVIOUS_TEST;
+    nextState = PREVIOUS_TEST;
+}
```
Qué sentido tiene esta función que sólo llamás una vez si, además, ya inicializaste las variables porque las hiciste globales ?

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,313 @@
+    char str[100];
+    int stringLength;
+    float vbatReading;
+    vbatReading = batVol.read();
+    vbatReading = 3.3 * vbatReading;
```
`float vbatReading = 3.3 * batVol.read();`

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,313 @@
+    float vbatReading;
+    vbatReading = batVol.read();
+    vbatReading = 3.3 * vbatReading;
+    sprintf(str, "Voltaje: %.2f V\r\n", vbatReading);
+    stringLength = strlen(str);
```
sprintf ya devuelve la cantidad de caracteres, HAY QUE LEER LOS MANUALES

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,313 @@
+    switch (current_State) {
+        case PREVIOUS_TEST:
+            secuenciaLed();
+            if(tecla_1.read() == 1){
+                    if(tecla_2.read() == 1){
```
`if((tecla_1.read() == 1) && (tecla_2.read() == 1) {`

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,313 @@
+                ledReady = OFF;
+                ledSystem = OFF;
+                probarBuzzer = true;
+            }else if(probarBuzzer == true){
+
+                if(timeOn < BUZZER_TIME_ON){
+                    timeOut = 0;
+                    outBuzzer = ON;
+                    timeOn = timeOn + TIME_INCREMENT_MS;
+                }
+                else{
```
else al costado
else abajo

Elegí un estilo y respetalo

scaprile commented on main.cpp:
```diff
@@ -0,0 +1,313 @@
+void secuenciaLed(){
+    static int accumulatedTime = 0;
+    accumulatedTime = accumulatedTime + TIME_INCREMENT_MS;
+    if(accumulatedTime == 750){
+		accumulatedTime = 0;
+	} else if(accumulatedTime < 250){
```
Siempre la misma cantidad de espacios o de tabs
Siemple espacios o siempre tabs

scaprile reviewed (approved):
Por fin entendí lo que querías hacer y lo pude seguir !

scaprile commented on README.md:
```diff
@@ -1,2 +1,82 @@
-# TP_1
-Primera entrega del TP_1 completo
+# CURSO ONLINE DE SISTEMAS EMBEBIDOS - Arm Book - Edisión 2024
```
edi**C**ión

nnzgab committed:
corregido error ortográfico en README.md

Corregido error ortográfico en README.md: se cambió "edisión" por "edición"

nnzgab committed:
Update main.cpp

se quita función redundante "void estadoInicial()"
se deja variable globales inicializadas

nnzgab committed:
Update main.cpp

Simplificado cálculo de vbatReading: combinado asignación y cálculo

nnzgab committed:
Update main.cpp

Combinado condiciones de tecla_1 y tecla_2 utilizando &&

nnzgab committed:
Update main.cpp

se acomoda tratando de seguir un solo estilo

nnzgab committed:
Update main.cpp

se aprovecha sprintf
